### ISSUE TITLE
OSError: [Errno 107] Transport endpoint is not connected

### ISSUE THREAD

**Issue by monajalal** [ai-tor/VPilot#9]
I get this error every once in a while randomly after I have the game in running state, and then execute the script `drive.py` on `Windows 10` Machine with  `Python 3.4.3`: 

```
mona@DESKTOP-37GGNHL:/mnt/d/aitor/VPilot$ python3 drive.py
Trying to connect to DeepGTAV
ERROR: Failed to connect to DeepGTAV
ERROR: Failed to send message. Reason: [Errno 107] Transport endpoint is not connected
Traceback (most recent call last):
  File "drive.py", line 42, in <module>
    message = client.recvMessage()
  File "/mnt/d/aitor/VPilot/deepgtav/client.py", line 49, in recvMessage
    frame = self._recvall()
  File "/mnt/d/aitor/VPilot/deepgtav/client.py", line 63, in _recvall
    packet = self.s.recv(4 - len(data))
OSError: [Errno 107] Transport endpoint is not connected
```
I'm not sure what's missing. Additionally, how could you set the driving mode to self-driving using the game AI similar to DeepGTAV1? I looked at `deepgtav/messages.py` as instructed in drive.py, but no further instruction was given there as well. Only -1 is shown to be manual driving. Can you tell what are the other options? I can have a wild guess of 0, 1 but would be great if you could add it to manual/code/etc. Thanks.

Other times, I get this:
```
mona@DESKTOP-37GGNHL:/mnt/d/aitor/VPilot$ python3 drive.py
Trying to connect to DeepGTAV
Successfully connected to DeepGTAV
```

Which because your saved games can't be loaded doesn't really do anything.  
The log for when it works:
```
Initializing Winsock...Initialized.
Socket created.
Bind done.
Listening...
Connection accepted.
Start received
Commands received
Commands received
Commands received
Commands received
Commands received
Commands received
Commands received
Commands received
Commands received

```

**Comment by monajalal** [comment]
So, as I guessed the problem goes back to not having the correct thing in my profile for saved games. I eventually used the one from DeepGTAV1 files. I have attached them here too. So the car has trottle itself but doesn't seem to be intelligent enough as it just crashes things on its way. DeepGTAV1 had an option to use Game AI. **_Can you please let us know how to get the game AI work?_**

https://drive.google.com/file/d/0B7j2NoupYo5gdTZDUVF6aFA5Mkk/view?usp=sharing


As seen above, in case VPilot and DeepGTAV2 are working correctly you should see the following in your deepgtav.log:
```
Initializing Winsock...Initialized.
Socket created.
Bind done.
Listening...
Connection accepted.
Start received
Commands received
Commands received
Commands received
etc
```

**Comment by aitorzip** [comment]
Hi Mona,

See dataset.py to see an example on how the environment can be configured so that the vehicle is driven by game's AI.

**Comment by tillmanp** [comment]
I am having the same issue as Mona. When I run drive.py it waits for about 5 minutes with "Trying to connect to DeepGTAV", then fails and shows the exact same error Mona posted above. It appears that the server is not starting. Is there something else I need to do to start the server? Thank you.

**Comment by tensorpro** [comment]
I have the same problem as tillmanp.

I wanted to make sure that the error wasn't due to network issues, so I ran my own server listening on port 8000. When I did this, my client was able to connect.

When I run my own server, netscan shows that my machine is listening on port 8000.
I've followed the instructions to install DeepGTAV, but running GTAV does not start a server listening on port 8000 for me.

Does DeepGTAV have debug logging I can see? 

EDIT:

I had that issue when I started the game and didn't select story mode.
After selecting story mode the server starts and listens on port 8000.
